BeVolt/Drivers/src/LTC6811.c

```diff
@@ -75,7 +75,185 @@ Copyright 2017 Linear Technology Corp. (LTC)
 /*********************************************************/
 /*** Code that was added by UTSVT. ***/
 /*********************************************************/
-void LTC6811_Init(cell_asic *battMod){	
+
+static void cs_set(uint8_t state){
+	if(state == 0){
+		GPIO_ResetBits(GPIOB, GPIO_Pin_6);
+	}else{
+		GPIO_SetBits(GPIOB, GPIO_Pin_6);
+	}
+}
+void delay_u(uint16_t micro){
+  uint32_t delay = SystemCoreClock / 1000000;
+	for(int i = 0; i < micro; i++){
+		for(int j = 0; j < delay; j++);
+	}
+}
+
+void delay_m(uint16_t milli){
+  uint32_t delay = SystemCoreClock / 1000;
+	for(int i = 0; i < milli; i++){
+		for(int j = 0; j < delay; j++);
+	}
+}
+
+/* wakeup_idle
+ * Wake isoSPI up from idle state
+ * @param number of ICs in the daisy chain
+ */
+void wakeup_idle(uint8_t total_ic){
+	volatile uint8_t tempReg;		// Temporary buffer
+  for (int i = 0; i<total_ic; i++){
+    cs_set(0);
+    //delayMicroseconds(2); //Guarantees the isoSPI will be in ready mode
+    tempReg = SPI1_Read();
+    cs_set(1);
+  }
+}
+
+/* wakeup_sleep
+ * Generic wakeup commannd to wake the LTC6811 from sleep
+ * @param number of ICs in the daisy chain
+ */
+void wakeup_sleep(uint8_t total_ic){
+  for (int i =0; i<total_ic; i++){
+    cs_set(0);
+    delay_u(300); // Guarantees the LTC6811 will be in standby
+    cs_set(1);
+    delay_u(10);
+  }
+}
+
+/* cmd_68
+ * Sense a command to the bms IC. This code will calculate the PEC code for the transmitted command
+ * @param 2 Byte array containing the BMS command to be sent
+ */ 
+void cmd_68(uint8_t tx_cmd[2]){
+  uint8_t cmd[4];
+  uint16_t cmd_pec;
+  //uint8_t md_bits;
+
+  cmd[0] = tx_cmd[0];
+  cmd[1] =  tx_cmd[1];
+  cmd_pec = pec15_calc(2, cmd);
+  cmd[2] = (uint8_t)(cmd_pec >> 8);
+  cmd[3] = (uint8_t)(cmd_pec & 0x00FF);
+	//cmd[3] = (uint8_t)(cmd_pec >> 8);
```

Comment:
This line should be removed to avoid confusion in the future, assuming that cmd[2] and cmd[3] are in fact calculated correctly above.